### PR TITLE
Add filter for severity

### PR DESCRIPTION
**Pull request by joris974**
Adding an optional list input allowing the client to only get notified on alerts from the given severity

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -32,6 +32,8 @@ inputs:
     description: 'SMTP user'
   email_transport_smtp_password:
     description: 'SMTP password'
+  severities:
+    description: 'List of severity to filter vulnerability alerts: LOW | MODERATE | HIGH | CRITICAL'
```
Comment (pbrisbin) [inline action.yml:36]:
```suggestion
    description: 'Newline-separated list of alert severities to notify on: LOW | MODERATE | HIGH | CRITICAL'
```

File: src/filters.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  alerts: Alert[],
+  severities: AdvisorySeverity[],
+) => {
+  if (severities.length === 0) {
+    return alerts;
+  }
+  return alerts.filter((alert) =>
+    alert.advisory !== undefined
+      ? severities.includes(alert.advisory.severity)
+      : true,
```
Comment (pbrisbin) [inline src/filters.ts:13]: So if there's no `alert.advisory`, but the user has specified `severities`, the alert is included? I wonder if it should be the opposite. If I've said to filter to severities, I think I'd expect an alert that doesn't have a severity at all to _not_ be included.

Either way, I would add a test case for this behavior.

File: src/main.ts [hunk 3]
```diff
@@ -29,30 +31,32 @@ async function run(): Promise<void> {
     const emailTransportSmtpUser = getInput('email_transport_smtp_user')
     const emailTransportSmtpPassword = getInput('email_transport_smtp_password')
     const count = parseInt(getInput('count'))
+    const severities = parseAdvisorySeverities(getMultilineInput('severities'))
     const owner = context.repo.owner
     const repo = context.repo.repo
     const alerts = await fetchAlerts(token, repo, owner, count)
-    if (alerts.length > 0) {
+    const filteredAlerts = filterAlertsByAdvisorySeverity(alerts, severities)
```
Comment (pbrisbin) [inline src/main.ts:38]: Not a huge deal, but you could save yourself from making additional diff noise by renaming the first variable instead,

```js
const allAlerts = await fetchAlerts(...)
const alerts = filterAlertsByAdvisorySeverity(...)
```

Or, alternatively, by doing the filtering in `fetchAlerts` instead.

**Review by pbrisbin — COMMENTED** [review]
What is your plan for these in-fork PRs vs PRs into upstream?

I usually just PR upstream right away and not worry about review from other Frecklers within the fork, but I'm not sure if you're requesting this review because you would like it or because you think it's some kind of process to do so. If it's the former, you probably want to find another Frontender. If it's the latter, I'm here to say, don't worry about it :)

File: src/filters.ts (hunk 2 above)
Comment (joris974) [reply src/filters.ts:13]: Both ways make sense. But I see how filtering them out are more aligned with the concept of filtering. I will make the change

**Comment by joris974** [comment]
I think your review is enough. I will check if this works and put another one for the path filtering

**Pull request merged**